scipy/io/tests/test_wavfile.py

```diff
@@ -75,36 +85,76 @@ def test_read_5():
         del data
 
 
-def test_read_fail():
+def test_read_unknown_filetype_fail():
+    # Not an RIFF
     for mmap in [False, True]:
-        fp = open(datafile('example_1.nc'), 'rb')
-        assert_raises(ValueError, wavfile.read, fp, mmap=mmap)
-        fp.close()
+        filename = 'example_1.nc'
+        with open(datafile(filename), 'rb') as fp:
+            with raises(ValueError) as excinfo:
+                wavfile.read(fp, mmap=mmap)
+
+            assert "CDF" in str(excinfo.value)
+            assert "'RIFF' and 'RIFX' supported" in str(excinfo.value)
+
+
+def test_read_unknown_riff_form_type():
+    # RIFF, but not WAVE form
+    for mmap in [False, True]:
+        filename = 'Transparent Busy.ani'
+        with open(datafile(filename), 'rb') as fp:
+            with raises(ValueError) as excinfo:
```

Comment:
I would just do `raises(ValueError, match='Not a WAV file.*ACON'):` where `match` is a suitable regex match. I usually try to match something that's specific enough to ensure we're catching the right thing but also not too long (no need to do the entire message).

Reply:
Ok, that makes them awkward though, like 

`with raises(ValueError, match='Unknown wave file format.*MULAW.*Supported formats'):`

I wanted to just test for a few components and allow the wording to be changed, etc. without affecting the test, but I don't know if that really matters

Reply:
> that makes them awkward though, like

That looks fine to me, quicker to read and more compact than the `excinfo`. I prefer it personally

> I wanted to just test for a few components and allow the wording to be changed, etc. without affecting the test, but I don't know if that really matters

If someone changes the wording such that it no longer matches it's fairly trivial to change the test to accommodate it, so I don't think we need to worry about this too much

Reply:
Ok I updated them